src/Microsoft.SqlTools.ServiceLayer/EditData/IEditTableMetadata.cs

```diff
@@ -0,0 +1,35 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.EditData
+{
+    /// <summary>
+    /// Interface for a metadata provider to be used for edit scenarios
```

Comment:
An <see cref="IEditTableMetadata" /> is used in edit data scenarios, defining necessary properties to determine which columns are editable, how to handle updates to the table, and what the primary key columns are among other things.